cypress/component/editor/search.cy.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * SPDX-FileCopyrightText: 2024 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { Editor } from '@tiptap/core'
+import { Document } from '@tiptap/extension-document'
+import { Text } from '@tiptap/extension-text'
+import Search from '../../../src/extensions/Search.js'
+import Paragraph from '../../../src/nodes/Paragraph.js'
+import HardBreak from '../../../src/nodes/HardBreak.js'
+
+describe('editor search highlighting', () => {
+	let editor = null
+
+	before(() => {
+		cy.fixture('lorem.txt').then((text) => {
+			editor = new Editor({
+				element: document.querySelector('div[data-cy-root]'),
+				content: text,
+				extensions: [Document, Text, Search, Paragraph, HardBreak],
+			})
+		})
+	})
+
+	it('can highlight a match', () => {
+		const searchQuery = 'Lorem ipsum dolor sit amet'
+		editor.commands.setSearchQuery(searchQuery)
+
+		const highlightedElements = document.querySelectorAll('span[data-text-el]')
```

Comment:
```suggestion
		const highlightedElements = document.querySelectorAll('span[data-text-el="search-decoration"]')
```

Nitpicking, but just to make sure to only catch search decoration elements, and not other `data-text-el` elements :wink: